daemon/cluster/utils.go

```diff
@@ -61,3 +65,40 @@ func removingManagerCausesLossOfQuorum(reachable, unreachable int) bool {
 func isLastManager(reachable, unreachable int) bool {
 	return reachable == 1 && unreachable == 0
 }
+
+// newErrWithHTTPCodeFromGRPC converts an error from gRPC into an error with HTTP status code
+func newErrWithHTTPCodeFromGRPC(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	switch grpc.Code(err) {
+	case codes.OK: // code 0
+		return nil
+	case codes.InvalidArgument: // code 3
+		return errors.NewBadRequestError(err)
```

Comment:
Thanks for your feedback. Like you mentioned, I tried to move that part into api/server/httputils/errors.go. PTAL
@cpuguy83